http-source/src/test/java/org/springframework/cloud/stream/module/http/DefaultHttpSourceTests.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.http;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.springframework.cloud.stream.test.matcher.MessageQueueMatcher.receivesPayloadThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.WebIntegrationTest;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.cloud.stream.test.binder.MessageCollector;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Tests for HttpSource with default mapping.
+ *
+ * @author Eric Bottard
+ * @author Mark Fisher
+ * @author Marius Bogoevici
+ * @author Ilayaperumal Gopinathan
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = HttpSourceApplication.class)
+public abstract class DefaultHttpSourceTests {
+
+	@Autowired
+	@Bindings(HttpSource.class)
+	protected Source channels;
+
+	@Value("${local.server.port}")
+	protected int port;
+
+	@Autowired
+	protected MessageCollector messageCollector;
+
+	protected RestTemplate restTemplate = new RestTemplate();
+
+	@WebIntegrationTest(randomPort = true)
+	public static class DefaultMappingTests extends DefaultHttpSourceTests {
```

Comment:
This one does use `randomPort` while the other one does not, so it's probably worth keeping... but as a follow-up, let's discuss the possibility of organizing the structure of these tests more clearly. I will go ahead and merge as-is.